tiledb/sm/query/ast/CMakeLists.txt

```diff
@@ -0,0 +1,131 @@
+#
+# tiledb/sm/query/ast/CMakeLists.txt
+#
+#
+# The MIT License
+#
+# Copyright (c) 2021-2022 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+include(common NO_POLICY_SCOPE)
+
+################################################################
+#
+# This is temporary until test_support is widely deployed
+#
+################################################################
+
+
+set(TILEDB_TEST_SUPPORT_SOURCES
+  ${CMAKE_CURRENT_SOURCE_DIR}/../../../../test/src/helpers.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/../../../../test/src/helpers.cc
+  ${CMAKE_CURRENT_SOURCE_DIR}/../../../../test/src/helpers-dimension.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/../../../../test/src/vfs_helpers.cc
+  ${CMAKE_CURRENT_SOURCE_DIR}/../../../../test/src/serialization_wrappers.cc
+  )
+
+
+message("Test support source " ${TILEDB_TEST_SUPPORT_SOURCES})
+
+add_library(tiledb_test_support_lib_ EXCLUDE_FROM_ALL
+  $<TARGET_OBJECTS:TILEDB_CORE_OBJECTS>
+  ${TILEDB_TEST_SUPPORT_SOURCES}
+)
+
+target_include_directories(tiledb_test_support_lib_
+  PRIVATE
+    $<TARGET_PROPERTY:Catch2::Catch2,INTERFACE_INCLUDE_DIRECTORIES>
+    target_include_directories(unit_ast PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/../../../../tiledb/sm/c_api")
```

Comment:
Why are we including the c_api for tests here? I don't see any of these includes needed?

Reply:
This entire file is mostly a CMake hack so that unit_ast can build.

Reply:
There was a dependency that was met this way.  I expect there is a better way -- I had asked about this in a sprint debugging session but a better way wasn't proffered at the time.  Part of the problem is the ordering of how object library test CMakeLists.txt files are included vs the actual unit test files -- the former are included before the latter so can't use them.  I will take another cut at making this more hygienic.

Reply:
The issue with including c_api is that the test helpers need tiledb.h and unit_ast needs the test helpers.  There is actually a lot of stuff that gets brought in with helpers that we don't really need for unit_ast.  Perhaps we should refactor just those things into ast_helpers?